dataflow/manual/content.tex

```diff
@@ -1735,6 +1735,32 @@ \section{Example: Live Variable}
 \flow{LiveSimple}{.33}{1}{Simple sequential program to illustrate live variable.  Intermediate analysis results are shown.}
 
 
+\section{Example: Reaching Definitions}
+
+A reaching definitions analysis for definitions(assignments) was implemented
+to show the forward analysis works properly. The main class is
+\code{org.checkerframework.dataflow.cfg.playground.ReachingDefinitionsPlayground}.
+
+\textbf{Abstract values.} A class \code{ReachingDefinitionsValue} is a reaching
+definitions (which is represented by a node) wrapper turning node into
+abstract value. A node can only be \code{AssignmentNode}.
+
+\textbf{The store.} The reaching definitions store \code{ReachingDefinitionsStore} has a field
+\code{reachingDefSet} which contains a set of \code{ReachingDefinitionsValue}s. Only
+\code{AssignmentNode} will be considered as a
+definition and added to \code{reachingDefSet}. The store defines methods
+\code{putDef(ReachingDefinitionValue)} and \code{killDef(Node)} to add
+and kill reaching definitions.
+
+\textbf{The transfer function.} The transfer function \code{ReachingDefinitionTransfer}
+initializes empty store. The transfer function visits assignments to
+update the information of reaching definitions for each node in the stores.
```

Comment:
Please briefly explan when to `put` and when to `kill`

Reply:
Okay!

Reply:
Did you address this in "store" above?

Reply:
No. Let me rewrite my document.